imxrt1062-rt/src/cache.rs

```diff
@@ -0,0 +1,88 @@
+//Cache configuration
+
+use core::ptr;
+
+static mut SCB_CCR: *mut u32 = 0xE000ED14 as *mut u32;
+static mut SCB_MPU_CTRL: *mut u32 = 0xE000ED94 as *mut u32;
+static mut SCB_MPU_RBAR: *mut u32 = 0xE000ED9C as *mut u32;
+static mut SCB_MPU_RASR: *mut u32 = 0xE000EDA0 as *mut u32;
+static mut SCB_CACHE_ICIALLU: *mut u32 = 0xE000EF50 as *mut u32;
+const SCB_CCR_IC: u32 = 1 << 17;
+const SCB_CCR_DC: u32 = 1 << 16;
+
+const SCB_MPU_RBAR_VALID: u32 = 1 << 4;
+const SCB_MPU_RASR_ENABLE: u32 = 1 << 0;
+const SCB_MPU_RASR_XN: u32 = 1 << 28;
+const fn scb_mpu_rasr_ap(n: u32) -> u32 {
+    (n & 7) << 24
+}
+const fn scb_mpu_rasr_tex(n: u32) -> u32 {
+    (n & 7) << 19
+}
+const SCB_MPU_RASR_C: u32 = 1 << 17;
+const SCB_MPU_RASR_B: u32 = 1 << 16;
+const SCB_MPU_CTRL_ENABLE: u32 = 1 << 0;
+const fn scb_mpu_rasr_size(n: u32) -> u32 {
+    (n & 31) << 1
+}
+const fn scb_mpu_rbar_region(n: u32) -> u32 {
+    n & 15
+}
+
+const NOEXEC: u32 = SCB_MPU_RASR_XN;
+const READONLY: u32 = scb_mpu_rasr_ap(7);
+const READWRITE: u32 = scb_mpu_rasr_ap(3);
+const _NOACCESS: u32 = scb_mpu_rasr_ap(0);
+const MEM_CACHE_WT: u32 = scb_mpu_rasr_tex(0) | SCB_MPU_RASR_C;
+const _MEM_CACHE_WB: u32 = scb_mpu_rasr_tex(0) | SCB_MPU_RASR_C | SCB_MPU_RASR_B;
+const MEM_CACHE_WBWA: u32 = scb_mpu_rasr_tex(1) | SCB_MPU_RASR_C | SCB_MPU_RASR_B;
+const MEM_NOCACHE: u32 = scb_mpu_rasr_tex(1);
+const DEV_NOCACHE: u32 = scb_mpu_rasr_tex(2);
+const SIZE_128K: u32 = scb_mpu_rasr_size(16) | SCB_MPU_RASR_ENABLE;
+const _SIZE_256K: u32 = scb_mpu_rasr_size(17) | SCB_MPU_RASR_ENABLE;
+const SIZE_512K: u32 = scb_mpu_rasr_size(18) | SCB_MPU_RASR_ENABLE;
+const SIZE_1M: u32 = scb_mpu_rasr_size(19) | SCB_MPU_RASR_ENABLE;
+const _SIZE_2M: u32 = scb_mpu_rasr_size(20) | SCB_MPU_RASR_ENABLE;
+const _SIZE_4M: u32 = scb_mpu_rasr_size(21) | SCB_MPU_RASR_ENABLE;
+const _SIZE_8M: u32 = scb_mpu_rasr_size(22) | SCB_MPU_RASR_ENABLE;
+const SIZE_16M: u32 = scb_mpu_rasr_size(23) | SCB_MPU_RASR_ENABLE;
+const _SIZE_32M: u32 = scb_mpu_rasr_size(24) | SCB_MPU_RASR_ENABLE;
+const SIZE_64M: u32 = scb_mpu_rasr_size(25) | SCB_MPU_RASR_ENABLE;
+const fn region(n: u32) -> u32 {
+    scb_mpu_rbar_region(n) | SCB_MPU_RBAR_VALID
+}
+
+extern "C" {
+    fn call();
+}
+
+#[inline(always)]
+pub unsafe fn init() {
```

Comment:
If it's not too much trouble, let's add a Rust doc comment with a `Safety` section. When we run `clippy`, the linter [yells at us about missing `Safety` documentation](https://rust-lang.github.io/rust-clippy/master/index.html#missing_safety_doc) on public, unsafe functions. Feel free to put something boring to make it pass, or add the appropriate `#[clippy(allow_xyz)]` directive.

```rust
//! # Safety
//! This function touches processor registers and should only be called once.
#[inline(always)
pub unsafe fn init() { ... }
```

(The code base doesn't pass all `clippy` checks just yet, but we intend on enabling it in CI soon.)

Reply:
Never mind! `clippy` doesn't seem to care about the missing "safety" documentation.